stackslib/src/net/http.rs

```diff
@@ -5047,6 +5143,321 @@ mod test {
     use crate::net::test::*;
     use crate::net::{RPCNeighbor, RPCNeighborsInfo};
 
+    /// Simulate reading variable-length segments
```

Comment:
Why are all these chunked HTTP transfer tests moved into this file?  Please put them back.

Reply:
I don't know what these are. There was a merge conflict here and maybe I didn't resolve it right. Where are these supposed to be?

Reply:
I see these were moved to `chunked_encoding.rs` and I ended up duplicating them here